### PR TITLE
feat(chart): Added additional service monitor functionality

### PR DESCRIPTION
**Pull request by stevehipwell**
### Issue

This PR will allow manual remediation of #2897.

### Description

This PR adds the ability to set the Helm chart `ServiceMonitor` endpoint values for `scrapeTimeout`, `relabelings` & `metricRelabelings`. It also fixes the `jobLabel` functionality (it should be a label name and not a value) and refactors the `ServiceMonitor` template for easier readability.

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [x] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by stevehipwell** [comment]
@kishorj @oliviassss could you please take a look at this PR? The failing workflow looks to have been failing for a while and the fix PR (#3600) was closed.

**Comment by stevehipwell** [comment]
@kishorj @oliviassss could you please take a look at this PR?

File: helm/aws-load-balancer-controller/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -29,7 +25,19 @@ spec:
   endpoints:
     - port: metrics-server
       path: /metrics
-      {{- with .Values.serviceMonitor.interval }}
+      scheme: http
```
Comment (oliviassss) [inline helm/aws-load-balancer-controller/templates/servicemonitor.yaml:28]: just for my understanding, why we specify `http` explicitly here?

Comment (stevehipwell) [reply helm/aws-load-balancer-controller/templates/servicemonitor.yaml:28]: I always set the `scheme` so it's very clear what the intent is even for someone who's starting at the `ServiceMonitor` and working up through the other resources. From an implementation point of view this is something where even though the default is correct we 100% care about the value, so we should explicitly set it.

**Comment by stevehipwell** [comment]
@oliviassss it looks like the dependency review job is currently failing across the board (also this PR isn't making any changes to the deps), are you planning on manually merging?

**Pull request merged**